app/javascript/packages/personal-key-input/index.js

```diff
@@ -0,0 +1,22 @@
+import base32Decode from 'base32-decode';
+import base32Encode from 'base32-encode';
+
+/**
+ * Coerce mistaken user input from 'problem' letters:
+ * https://en.wikipedia.org/wiki/Base32#Crockford.27s_Base32
+ *
+ * @param {string} value User-provided text input
+ *
+ * @return {string} Encoded input
+ */
+export function encodeInput(value) {
+  value = value.replace(/-/g, '');
+  value = base32Decode(value, 'Crockford');
+  value = base32Encode(value, 'Crockford');
```

Comment:
TypeScript is assuming based on your type annotations that `value` is a string. It sees the return value of `base32Decode` is an `ArrayBuffer`. Since this isn't assignable to string, it becomes upset. And it isn't smart enough to see ahead to the fact that you eventually re-encode it.

Options:

- Create a different variable for `const decodedValue`.
- Inline the two-step decode and encode:

```suggestion
  value = base32Encode(base32Decode(value, 'Crockford'), 'Crockford');
```